File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotEmptyRuleTest.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.StringRes;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Locale;

import static org.junit.Assert.*;

public class NotEmptyRuleTest {

    private NotEmptyRule notEmptyRule;

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
        notEmptyRule = null;
    }

    @Test(expected = NullPointerException.class)
    public void validate_EmptySample_ThrowNullPointerException() throws Exception {
        notEmptyRule = new NotEmptyRule();

        notEmptyRule.validate(null);
    }

    @Test
    public void validate_ValidSample_ReturnTrue() throws Exception {
        notEmptyRule = new NotEmptyRule();

        String sample = "not empty";

        assertTrue(notEmptyRule.validate(sample));
    }

    @Test
    public void validate_InvalidSample_ReturnFalse() throws Exception {
        notEmptyRule = new NotEmptyRule();

        String sample = "";

        assertFalse(notEmptyRule.validate(sample));
    }

    @Test
    public void errorMessage_DefaultErrorMessage() throws Exception {
        String errorMessage = "Value must not be empty";

        notEmptyRule = new NotEmptyRule();

        assertEquals(errorMessage, notEmptyRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorMessage() throws Exception {
        String errorMessage = "This is a custom error message";

        notEmptyRule = new NotEmptyRule(errorMessage);

        assertEquals(errorMessage, notEmptyRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorRes() throws Exception {
        @StringRes int errorRes = 0;

        notEmptyRule = new NotEmptyRule(errorRes);

        assertEquals(errorRes, notEmptyRule.errorRes());
    }
}

File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotEmptyRuleTest.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import org.junit.After
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class NotEmptyRuleTest {

    private lateinit var notEmptyRule: NotEmptyRule

    @Before
    @Throws(Exception::class)
    fun setUp() {
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @Test(expected = NullPointerException::class)
    @Throws(Exception::class)
    fun validate_EmptySample_ThrowNullPointerException() {
        notEmptyRule = NotEmptyRule()

        notEmptyRule.validate(null)
    }

    @Test
    @Throws(Exception::class)
    fun validate_StringSample_ReturnTrue() {
        notEmptyRule = NotEmptyRule()

        val sample = "not empty"

        assertTrue(notEmptyRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_StringSample_ReturnFalse() {
        notEmptyRule = NotEmptyRule()

        val sample = ""

        assertFalse(notEmptyRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_SpaceSample_ReturnTrue() {
        notEmptyRule = NotEmptyRule()

        val sample = " "

        assertTrue(notEmptyRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_DefaultErrorMessage() {
        val errorMessage = "Value must not be empty"

        notEmptyRule = NotEmptyRule()

        assertEquals(errorMessage, notEmptyRule.getErrorMessage())
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorMessage() {
        val errorMessage = "This is a custom error message"

        notEmptyRule = NotEmptyRule(errorMessage)

        assertEquals(errorMessage, notEmptyRule.getErrorMessage())
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorRes() {
        @StringRes val errorRes = 0

        notEmptyRule = NotEmptyRule(errorRes)

        assertEquals(errorRes, notEmptyRule.getErrorRes())
    }
}